lib/widgets/new_dm_sheet.dart

```diff
@@ -0,0 +1,327 @@
+        .toList()
+        ..sort((a, b) {
+          final recentDms = store.recentDmConversationsView;
+          final aLatestMessageId = recentDms.latestMessagesByRecipient[a.userId];
+          final bLatestMessageId = recentDms.latestMessagesByRecipient[b.userId];
+
+          if (aLatestMessageId != null && bLatestMessageId != null) {
+            return bLatestMessageId.compareTo(aLatestMessageId);
+          }
+          if (aLatestMessageId != null) return -1;
+          if (bLatestMessageId != null) return 1;
+          return 0;
+        });
+    });
+  }
+
+  // Scroll to the search field when the user selects a user
+  void scrollToSearchField() {
+    SchedulerBinding.instance.addPostFrameCallback((_) {
+      if (scrollController.hasClients) {
+        scrollController.animateTo(
+          scrollController.position.maxScrollExtent,
+          duration: const Duration(milliseconds: 200),
+          curve: Curves.easeOut);
+      }
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return SizedBox(
+      height: MediaQuery.of(context).size.height * 0.95,
+      child: Column(children: [
+        NewDmHeader(
+          selectedUserIds: selectedUserIds,
+          onClose: () => Navigator.pop(context)),
+        NewDmSearchBar(
+          searchController: searchController,
+          scrollController: scrollController,
+          selectedUserIds: selectedUserIds),
+        Expanded(child: NewDmUserList(
+          filteredUsers: filteredUsers,
+          selectedUserIds: selectedUserIds,
+          onUserSelected: (userId) {
+            setState(() {
+              if (selectedUserIds.contains(userId)) {
+                selectedUserIds.remove(userId);
+              } else {
+                selectedUserIds.add(userId);
+              }
+              scrollToSearchField();
+            });
+          }))]));
+  }
+}
+
+class NewDmHeader extends StatelessWidget {
+  const NewDmHeader({
+    super.key,
+    required this.selectedUserIds,
+    required this.onClose});
+
+  final Set<int> selectedUserIds;
+  final VoidCallback onClose;
+
+  @override
+  Widget build(BuildContext context) {
+    final designVariables = DesignVariables.of(context);
+    final store = PerAccountStoreWidget.of(context);
+    final zulipLocalizations = ZulipLocalizations.of(context);
+
+    return Container(
+      padding: const EdgeInsets.all(8),
+      child: Row(
+        mainAxisAlignment: MainAxisAlignment.spaceBetween,
+        children: [
+          _buildBackButton(context, designVariables, zulipLocalizations),
+          Text(zulipLocalizations.newDmSheetScreenTitle,
+            style: TextStyle(color: designVariables.title,fontSize: 20, fontWeight: FontWeight.w600)),
+          _buildNextButton(context, store, designVariables, zulipLocalizations)]));
+  }
+
+  Widget _buildBackButton(BuildContext context, DesignVariables designVariables,
+      ZulipLocalizations zulipLocalizations) => TextButton(
```

Comment:
If the helpers already have access to a `BuildContext`, there is generally no need to pass `designVariables` etc around.

Reply:
Since this method is already pretty long, we should [use braces](https://github.com/flutter/flutter/blob/master/docs/contributing/Style-guide-for-Flutter-repo.md#use-braces-for-long-functions-and-methods).

Reply:
I think it might actually be cleaner to extract these helpers as local variables in `NewDmHeader.build`.

If you look at the other `Widget _buildSomething` style helpers in the code base with `git grep -A2 'Widget _build'`, you may find that they often have other parameters not tied to the `BuildContext`.